specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/CommonDefinitions.json

```diff
@@ -1614,6 +1614,208 @@
+      }
+    },
+    "ManagedHttpRoute": {
+      "description": "Managed certificates used for Custom Domain bindings of Container Apps in a Managed Environment",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Certificate resource specific properties",
+          "type": "object",
+          "properties": {
+            "provisioningState": {
+              "description": "The scale configuration of the session pool.",
+              "$ref": "#/definitions/HttpRouteProvisioningState"
+            },
+            "fqdn": {
+              "description": "FQDN of the route resource.",
+              "type": "string",
+              "readOnly": true
+            },
+            "customDomains": {
+              "description": "custom domain bindings for http Routes' hostnames.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/CustomDomain"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "rules": {
+              "description": "Routing Rules for http route resource.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/HttpRouteRule"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            }
+          }
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "CustomDomain": {
+      "description": "Custom Domain of a Container App",
+      "type": "object",
+      "required": [
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "Hostname.",
+          "type": "string"
+        },
+        "bindingType": {
+          "description": "Custom Domain binding type.",
+          "enum": [
+            "Disabled",
+            "SniEnabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "bindingType",
+            "modelAsString": true
+          }
+        },
+        "certificateId": {
+          "description": "Resource Id of the Certificate to be bound to this hostname. Must exist in the Managed Environment.",
+          "type": "string"
+        }
+      }
+    },
+    "HttpRouteTarget": {
+      "description": "Targets - Container App Names, Revision Names, Labels.",
+      "type": "object",
+      "properties": {
+        "containerAppName": {
+          "description": "Container App Name to route requests to",
+          "type": "string"
+        },
+        "revisionName": {
+          "description": "Revision to route requests to",
+          "type": "string"
+        },
+        "labelName": {
+          "description": "Label/Revision to route requests to",
+          "type": "string"
+        },
+        "weight": {
```

Comment:
>weight [](http://example.com/codeflow?start=9&length=6)

are there limits to this prop? can you add negative values?

Reply:
No cannot be negative. Sum of values need to add upto 100

Reply:
done